src/test/java/test/org/jenkinsci/plugins/workinghours/WorkingHoursQueueTaskDispatcherTest.java

```diff
@@ -186,7 +188,139 @@ public void testCanRunBlockedNotPlaceholder() {
+        when(task.run()).thenReturn(run);
+        when(job.getProperty(EnforceScheduleJobProperty.class)).thenReturn(prop);
+        when(prop.getBranches()).thenReturn(new ArrayList<>());
+
+        WorkingHoursQueueTaskDispatcher instance = new WorkingHoursQueueTaskDispatcher();
+        assertNotNull(instance.canRun(item));
+    }
+
+    /**
+     * Verifies that canRun returns a blockage reason when not a MultibranchPipeline
+     */
+    @Test
+    public void testCanRunBlockedNonMultibranchPipeline() {
+        WorkingHoursConfig config = WorkingHoursConfig.get();
+        config.setBuildTimeMatrix(TimeRangeUtility.getExclusiveRange());
+        config.save();
+
+        ExecutorStepExecution.PlaceholderTask task = mock(ExecutorStepExecution.PlaceholderTask.class);
+        Queue.WaitingItem item = new Queue.WaitingItem(Calendar.getInstance(), task, Collections.EMPTY_LIST);
+
+        EnforceScheduleJobProperty prop = mock(EnforceScheduleJobProperty.class);
+        WorkflowJob job = mock(WorkflowJob.class);
+        Run run = mock(Run.class);
+
+        when(task.getOwnerTask()).thenReturn(job);
+        when(task.run()).thenReturn(run);
+        when(job.getProperty(EnforceScheduleJobProperty.class)).thenReturn(prop);
+        when(job.getParent()).thenReturn(mock(Hudson.class));
+        when(prop.getBranches()).thenReturn(new ArrayList<>(Arrays.asList("TestBranch")));
+
+        WorkingHoursQueueTaskDispatcher instance = new WorkingHoursQueueTaskDispatcher();
+        assertNotNull(instance.canRun(item));
+    }
+
+
     /*
+     * Verifies that canRun returns a blockage reason when current branch doesn't match branches provided
+     *
+    @Test
+    public void testCanRunBlockedNonMatchingBranches() {
+        WorkingHoursConfig config = WorkingHoursConfig.get();
+        config.setBuildTimeMatrix(TimeRangeUtility.getExclusiveRange());
+        config.save();
+
+        ExecutorStepExecution.PlaceholderTask task = mock(ExecutorStepExecution.PlaceholderTask.class);
+        Queue.WaitingItem item = new Queue.WaitingItem(Calendar.getInstance(), task, Collections.EMPTY_LIST);
+
+        EnforceScheduleJobProperty prop = mock(EnforceScheduleJobProperty.class);
+        WorkflowJob job = mock(WorkflowJob.class);
+        Run run = mock(Run.class);
+
+        when(task.getOwnerTask()).thenReturn(job);
+        when(task.run()).thenReturn(run);
+        when(job.getProperty(EnforceScheduleJobProperty.class)).thenReturn(prop);
+        doReturn(mock(WorkflowMultiBranchProject.class)).when(job).getParent();
+        when(job.getDisplayName()).thenReturn("master");
+        when(prop.getBranches()).thenReturn(new ArrayList<>(Arrays.asList("TestBranch")));
+
+        WorkingHoursQueueTaskDispatcher instance = new WorkingHoursQueueTaskDispatcher();
+        assertNull(instance.canRun(item));
+    }
+
+    /**
+     * Verifies that canRun blocks tasks when current branch matches a branch provided and is a MultibranchPipeline
+     *
```

Comment:
I feel like a test that verifies it doesn't block when the current branch matches a branch provided, but it's outside of working hours would be useful as well.